private/model/api/operation.go

```diff
@@ -162,43 +163,35 @@ func (c *{{ .API.StructName }}) {{ $reqType }}(input {{ .InputRef.GoType }}) ({{
 //            return pageNum <= 3
 //        })
 //
-func (c *{{ .API.StructName }}) {{ .ExportedName }}Pages(` +
-	`input {{ .InputRef.GoType }}, fn func({{ .OutputRef.GoType }}, bool) bool) error {
-	return c.{{ .ExportedName }}PagesWithContext(aws.BackgroundContext(), input, fn)
-}
-
-// {{ .ExportedName }}PagesWithContext same as {{ .ExportedName }}Pages except
-// it takes a Context and allows setting request options on the pages.
-//
-// The context must be non-nil and will be used for request cancellation. If
-// the context is nil a panic will occur. In the future the SDK may create
-// sub-contexts for http.Requests. See https://golang.org/pkg/context/
-// for more information on using Contexts.
-func (c *{{ .API.StructName }}) {{ .ExportedName }}PagesWithContext(` +
-	`ctx aws.Context, ` +
-	`input {{ .InputRef.GoType }}, ` +
-	`fn func({{ .OutputRef.GoType }}, bool) bool, ` +
-	`opts ...aws.Option) error {
-	p := aws.Pagination {
-		NewRequest: func() (*aws.Request, error) {
-			var inCpy {{ .InputRef.GoType }}
-			if input != nil  {
-				tmp := *input
-				inCpy = &tmp
-			}
-			req := c.{{ .ExportedName }}Request(inCpy)
-			req.SetContext(ctx)
-			req.ApplyOptions(opts...)
-			return req.Request, nil
+func (p *{{ $reqType }}) Paginate(opts ...aws.Option) {{ $pagerType }} {
+	return {{ $pagerType }}{
+			aws.Pager {NewRequest: func() (*aws.Request, error) {
+				var inCpy {{ .InputRef.GoType }}
+				if p.Input != nil  {
+					tmp := *p.Input
+					inCpy = &tmp
+				}
+
+				var output {{ .OutputRef.GoTypeElem }}
+				req := aws.New(p.Request.Config, p.Request.Metadata, p.Request.Handlers.Copy(), p.Request.Retryer, p.Request.Operation, inCpy, &output)
```

Comment:
If there are in custom code generated into the API's __Request method this will bypass that code. e.g. Signing logic, ect.  

For example its possible for the __Request logic to have a closure referencing some instance value such as a output shape. This code probably should just use the base __Request instead of bypasing it.

Reply:
Okay, that makes sense.